src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Portal/New-ArmDashboardsWebTestsPinned.ps1

```diff
@@ -0,0 +1,62 @@
+function New-ArmDashboardsWebTestsPinned {
+  [CmdletBinding(SupportsShouldProcess = $true)]
+  [OutputType("DashboardPart")]
+  Param(
+    [PSTypeName("ApplicationInsights")]
+    [Parameter(Mandatory)]
+    $ApplicationInsights,
+    [string]
+    [Parameter(Mandatory)]
+    $SubscriptionId,
+    [string]
+    [Parameter(Mandatory)]
+    $ResourceGroupName
+  )
+
+  If ($PSCmdlet.ShouldProcess("Adding WebTestsPinnedPart to Dashboards")) {
+    $ApplicationInsightsResourceName = $ApplicationInsights.Name
+    $WebTests = [PSCustomObject][ordered]@{
+      PSTypeName = "DashboardPart"
+      position   = @{ }
+      metadata   = @{
+        inputs = @(@{
+            name  = 'ComponentId'
+            value = @{
+              Name           = $ApplicationInsightsResourceName
+              SubscriptionId = $SubscriptionId
+              ResourceGroup  = $ResourceGroupName
+            }
+          }, 
+          @{
```

Comment:
The TimeContext is also reused at a lot of place
```
          @{
            name  = 'TimeContext'
            value = @{
              durationMs            = 86400000
              endTime               = $null
              createdTime           = '2018-05-04T01:20:33.345Z'
              isInitialTime         = $true
              grain                 = 1
              useDashboardTimeRange = $false
            }
          }
```
Is it possible to extract it?